test/system/src/test/java/io/pravega/test/system/ConsumptionBasedRetentionWithMultipleReaderGroupsTest.java

```diff
@@ -280,7 +242,185 @@ public void multipleSubscriberCBRTest() throws Exception {
+                EventWriterConfig.builder().build());
+
+        // Write 7 events to the stream_1.
+        writingEventsToStream(7, writer, SCOPE_1, STREAM_1);
+        // Write 10 events to the stream_2.
+        writingEventsToStream(10, writer2, SCOPE_1, STREAM_2);
+
+        @Cleanup
+        ReaderGroupManager readerGroupManager = ReaderGroupManager.withScope(SCOPE_1, clientConfig);
+        ReaderGroupConfig readerGroupConfig = getReaderGroupConfig(SCOPE_1, STREAM_1, ReaderGroupConfig.StreamDataRetention.MANUAL_RELEASE_AT_USER_STREAMCUT);
+        ReaderGroupConfig readerGroupConfig2 = getReaderGroupConfig(SCOPE_1, STREAM_2, ReaderGroupConfig.StreamDataRetention.MANUAL_RELEASE_AT_USER_STREAMCUT);
+
+        assertTrue("Reader group is not created", readerGroupManager.createReaderGroup(READER_GROUP_3, readerGroupConfig));
+        assertTrue("Reader group is not created", readerGroupManager.createReaderGroup(READER_GROUP_4, readerGroupConfig2));
+        assertEquals(1, controller.listSubscribers(SCOPE_1, STREAM_1).join().size());
+        assertEquals(1, controller.listSubscribers(SCOPE_1, STREAM_2).join().size());
+
+        @Cleanup
+        ReaderGroup readerGroup = readerGroupManager.getReaderGroup(READER_GROUP_3);
+        @Cleanup
+        ReaderGroup readerGroup2 = readerGroupManager.getReaderGroup(READER_GROUP_4);
+        AtomicLong clock = new AtomicLong();
+        @Cleanup
+        EventStreamReader<String> reader = clientFactory.createReader(READER_GROUP_3 + "-" + 1,
+                READER_GROUP_3, new JavaSerializer<>(), readerConfig, clock::get, clock::get);
+        @Cleanup
+        EventStreamReader<String> reader2 = clientFactory.createReader(READER_GROUP_4 + "-" + 1,
+                READER_GROUP_4, new JavaSerializer<>(), readerConfig, clock::get, clock::get);
+
+        // Read three events with reader.
+        readingEventsFromStream(3, reader);
+        // Read five events with reader2.
+        readingEventsFromStream(5, reader2);
+
+        log.info("{} generating 1st stream-cuts for {}/{}", READER_GROUP_3, SCOPE_1, STREAM_1);
+        Map<Stream, StreamCut> streamCuts = generateStreamCuts(readerGroup, reader, clock);
+        log.info("{} generating 1st stream-cuts for {}/{}", READER_GROUP_4, SCOPE_1, STREAM_2);
+        Map<Stream, StreamCut> streamCuts2 = generateStreamCuts(readerGroup2, reader2, clock);
+
+        log.info("{} updating its retention stream-cut to {}", READER_GROUP_3, streamCuts);
+        readerGroup.updateRetentionStreamCut(streamCuts);
+        log.info("{} updating its retention stream-cut to {}", READER_GROUP_4, streamCuts2);
+        readerGroup2.updateRetentionStreamCut(streamCuts2);
+
+        // Retention set has one stream cut at 0/210
+        // READER_GROUP_3 updated stream cut at 0/90
+        // Subscriber lower bound is 0/90, truncation should happen at this point
+        // The timeout is set to 3 minutes a little longer than the retention period which is set to 2 minutes
```

Comment:
Need to update this comment line. Now timeout is set to 2 mins and retention period is 1 min.

Reply:
done